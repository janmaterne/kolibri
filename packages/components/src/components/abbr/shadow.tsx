import type { JSX } from '@stencil/core';
import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
import type { AbbrAPI, AbbrStates, LabelPropType } from '../../schema';
import { validateLabel } from '../../schema';

/**
 * @slot - The abbreviation (short form).
 */
@Component({
	tag: 'kol-abbr',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolAbbr implements AbbrAPI {
	public render(): JSX.Element {
		return (
			<Host class="kol-abbr">
				<abbr>
					<slot />
				</abbr>
				{this.state._label ? ` (${this.state._label})` : ''}
			</Host>
		);
	}

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label?: LabelPropType;

	@State() public state: AbbrStates = {
		_label: '', // âš  required
	};

	@Watch('_label')
	public validateLabel(value?: LabelPropType): void {
		validateLabel(this, value, {
			required: true,
		});
	}

	public componentWillLoad(): void {
		this.validateLabel(this._label);
	}
}
