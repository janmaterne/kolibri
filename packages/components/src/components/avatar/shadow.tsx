import type { JSX } from '@stencil/core';
import { Component, h, Prop } from '@stencil/core';

import type { AvatarProps } from '../../schema';
import { KolAvatarWcTag } from '../../core/component-names';

@Component({
	tag: 'kol-avatar',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolAvatar implements AvatarProps {
	public render(): JSX.Element {
		return <KolAvatarWcTag _src={this._src} _label={this._label}></KolAvatarWcTag>;
	}

	/**
	 * Sets the image `src` attribute to the given string.
	 */
	@Prop() public _src?: string;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: string;
}
