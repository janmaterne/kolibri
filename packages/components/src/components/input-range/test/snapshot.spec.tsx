import { KolInputRangeTag } from '@component-names';
import type { InputRangeProps } from '@schema';
import { executeInputSnapshotTests } from '@testing';

import { KolInputRange } from '../shadow';

executeInputSnapshotTests<InputRangeProps>(KolInputRangeTag, [KolInputRange], {
	_value: 5,
	_min: 1,
	_max: 10,
	_step: 1,
});

executeInputSnapshotTests<InputRangeProps>(KolInputRangeTag, [KolInputRange], {
	_value: 5,
	_min: 1,
	_max: 10,
	_step: 1,
	_suggestions: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
});
