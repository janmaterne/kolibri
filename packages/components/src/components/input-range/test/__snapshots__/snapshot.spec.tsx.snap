// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _accessKey="V" 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _accesskey="V" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          V
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input accesskey="V" aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _alert=true 1`] = `
<kol-input-range _alert="" _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _alert="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _disabled=true 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _disabled="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" disabled="" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" disabled="" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _hint="Hint" 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="Hint" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-describedby="id-nonce-hint" aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-input-range _hide-error="" _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hideerror="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _readOnly=true 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" _readonly="" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _accessKey="V" 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _accesskey="V" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          V
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input accesskey="V" aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _alert=true 1`] = `
<kol-input-range _alert="" _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _alert="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _disabled=true 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _disabled="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" disabled="" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" disabled="" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _hint="Hint" 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="Hint" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-describedby="id-nonce-hint" aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-input-range _hide-error="" _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hideerror="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _readOnly=true 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" _readonly="" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] _touched=true 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" _touched="" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _alert="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" _touched="" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _suggestions=[1,2,3,4,5,6,7,8,9,10] 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" list="id-nonce-list" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
          <option value="4"></option>
          <option value="5"></option>
          <option value="6"></option>
          <option value="7"></option>
          <option value="8"></option>
          <option value="9"></option>
          <option value="10"></option>
        </datalist>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 _touched=true 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" _touched="" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _alert="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" _touched="" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;

exports[`kol-input-range should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value=5 _min=1 _max=10 _step=1 1`] = `
<kol-input-range _placeholder="Hier steht ein Platzhaltertext" class="kol-input-range">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="range">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <div class="inputs-wrapper" style="--kolibri-input-range--input-number--width: 2.5em;">
          <input aria-hidden="true" autocapitalize="off" autocomplete="off" autocorrect="off" max="10" min="1" name="field-range" spellcheck="false" step="1" tabindex="-1" type="range" value="5">
          <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="10" min="1" name="field-number" step="1" type="number" value="5">
        </div>
      </div>
    </kol-input>
  </template>
</kol-input-range>
`;
